src/query.ts

```diff
@@ -0,0 +1,49 @@
+import {
+    QueryParams,
+    QueryParamsPositional,
+    isQueryParamsNamed,
+    isQueryParamsPositional,
+} from './query-params'
+
+export type Query = {
+    query: string
+    parameters?: QueryParams
+}
+
+export function constructRawQuery(query: Query) {
+    if (isQueryParamsNamed(query.parameters)) {
+        // Replace named parameters with the actual values
+        let queryWithParams = query.query
+
+        for (const [key, value] of Object.entries(query.parameters)) {
+            if (typeof value === 'string') {
+                queryWithParams = queryWithParams.replace(
+                    `:${key}`,
+                    `'${value}'`
+                )
+            } else {
+                queryWithParams = queryWithParams.replace(`:${key}`, value)
+            }
+        }
+
+        return queryWithParams
+    } else if (isQueryParamsPositional(query.parameters)) {
```

Comment:
Doesn't need to be an else if. You added a return in the previous statement.

Reply:
Also a lot of boilerplate here. You can remove your // comments by naming a function that does what the comment is explaing. `insertPositionalArguments()` or something like that maybe. It would clean this function up and make it easier to understand what `constructRawQuery` is doing without having implementation logic within it.

Reply:
I left it as an `if/else if` just in case for some reason the `query.parameters` was undefined then it wouldn't go into either. But that's a bug, now I have the first one catch if it's `undefined` and let it channel through there safely.